src/uwtools/tests/test_schemas.py

```diff
@@ -90,6 +95,140 @@ def test_schema_chgres_cube_run_dir(chgres_cube_prop):
     assert "88 is not of type 'string'" in errors(88)
 
 
+# esg-grid
+
+
+def test_schema_esg_grid():
+    config = {
+        "execution": {"executable": "esg_grid"},
+        "run_dir": "/tmp",
+    }
+    errors = schema_validator("esg-grid", "properties", "esg_grid")
+    # Basic correctness:
+    assert not errors(config)
+    # Some top-level keys are required:
+    for key in ("execution", "run_dir"):
+        assert f"'{key}' is a required property" in errors(with_del(config, key))
+    # Additional top-level keys are not allowed:
+    assert "Additional properties are not allowed" in errors({**config, "foo": "bar"})
+    # Additional MPI support is not allowed:
+    assert "Additional properties are not allowed ('mpicmd' was unexpected)" in errors(
+        {"execution": {"mpicmd": "srun"}}
+    )
+
+
+def test_schema_esg_grid_namelist(esg_grid_prop):
+    base_file = {"base_file": "/some/path"}
+    update_values = {
+        "update_values": {
+            "regional_grid_nml": {
+                "delx": 0.22,
+                "dely": 0.22,
+                "lx": -200,
+                "ly": -130,
+                "pazi": 0.0,
+                "plat": 45.5,
+                "plon": -100.5,
+            }
+        }
+    }
+    errors = esg_grid_prop("namelist")
+    # Just base_file is ok:
+    assert not errors(base_file)
+    # base_file must be a string:
+    assert "{'base_file': 88} is not valid under any of the given schemas" in errors(
+        {"base_file": 88}
+    )
+    # Just update_values is ok:
+    assert not errors(update_values)
+    # A combination of base_file and update_values is ok:
+    assert not errors({**base_file, **update_values})
+    # All key/value pairs in update_values must be present if base_file is not supplied:
+    # pylint: disable=line-too-long
+    assert (
+        "{'update_values': {'regional_grid_nml': {'delx': 0.11, 'lx': -180, 'plat': 38.0}}} is not valid under any of the given schemas"
+        in errors(
+            {"update_values": {"regional_grid_nml": {"delx": 0.11, "lx": -180, "plat": 38.0}}}
+        )
+    )
```

Comment:
I did run into trouble trying to use `with_del`. When I tried to do
`"{'update_values': {}} is not valid under any of the given schemas in {'base_file': 'some/str', 'update_values': {}} is not valid under any of the given schemas" in "{'base_file': 'some/str', 'update_values': {}} is not valid under any of the given schemas`

I was having a problem with this second `in` without the assert statement because it was showing up as a special character. So I went about it this way to avoid that.

Reply:
I tried your suggestion of:

> I also think that a check for just is not valid under any of the given schemas would be fine here, without the detail

That worked great so I'll simplify that!